### PR TITLE
add various `ptp_*` structs

### PR DESCRIPTION
**Pull request by folkertdev**
# Description

Just the structs (i.e. skipping some constants) from https://github.com/rust-lang/libc/pull/3865. Hopefully this will be easier to get through CI, and then we can look at the constants.

# Sources

# Checklist

- [x] Relevant tests in `libc-test/semver` have been updated
- [x] No placeholder or unstable values like `*LAST` or `*MAX` are
  included (see [#3131](https://github.com/rust-lang/libc/issues/3131))
- [ ] Tested locally (`cd libc-test && cargo test --target mytarget`);
  especially relevant for platforms that may not be checked in CI (sadly that just does not compile)

r? @tgross35

### CODE REVIEW

File: src/unix/linux_like/linux/mod.rs [hunk 1]
```diff
@@ -4464,6 +4565,19 @@ pub const HWTSTAMP_FILTER_PTP_V2_SYNC: ::c_uint = 13;
...
+
+pub const PTP_CLOCK_CAPS_RSV_LEN: usize = {
+    if cfg!(any(target_arch = "sparc", target_arch = "sparc64")) {
+        12
+    } else if cfg!(any(target_env = "musl", target_env = "ohos")) {
+        13
+    } else {
+        11
+    }
+};
```
Comment (folkertdev) [inline src/unix/linux_like/linux/mod.rs:4579]: note: `cfg_if!` does not work here; the const symbol just does not show up in the C tests.

File: src/unix/linux_like/linux/mod.rs [hunk 2]
```diff
@@ -19,6 +19,7 @@ pub type pthread_key_t = ::c_uint;
 pub type pthread_once_t = ::c_int;
 pub type pthread_spinlock_t = ::c_int;
 pub type __kernel_fsid_t = __c_anonymous__kernel_fsid_t;
+pub type __kernel_clockid_t = ::c_int;
```
Comment (folkertdev) [inline src/unix/linux_like/linux/mod.rs:22]: the `__kernel_fsid_t` symbol is not in the semver file, so I didn't add `__kernel_clockid_t` either. But maybe they should both be added?

**Comment by tgross35** [comment]
```
  --- stderr
  error: expected `*` or `+`
     --> ../src/macros.rs:248:87
      |
  248 |                 pub $({$constness:ident})* fn $i:ident($($arg:ident: $argty:ty),* $(,)?) -> $ret:ty
      |  
```

I have no clue why this is showing up in CI here but I am so sick of ctest. Feel free to add a commit that changes the `$(,)?` to `$(,)*` with a`FIXME(ctest)` if it's needed to unblock this.

File: src/unix/linux_like/linux/mod.rs (hunk 2 above)
Comment (tgross35) [reply src/unix/linux_like/linux/mod.rs:22]: Would you be able to add both? Even though they're dunder types, it's nice to at least make it clear when they disappear.

Comment (folkertdev) [reply src/unix/linux_like/linux/mod.rs:22]: sure, I added them

File: src/unix/linux_like/linux/mod.rs (hunk 1 above)
Comment (folkertdev) [reply src/unix/linux_like/linux/mod.rs]: I just removed this const and the struct that used it for now.

**Comment by folkertdev** [comment]
I just removed the `ptp_clock_caps` struct for now, and that passes CI. So, I think this is good now?

**Review by tgross35 — APPROVED** [review]
Hey look, passing CI!

LGTM, could you squash?

**Comment by folkertdev** [comment]
wow, is this stuff cursed or something?

```
0.757 Reading package lists...
0.766 E: The repository 'http://security.ubuntu.com/ubuntu mantic-security Release' does not have a Release file.
0.766 E: The repository 'http://archive.ubuntu.com/ubuntu mantic Release' does not have a Release file.
0.766 E: The repository 'http://archive.ubuntu.com/ubuntu mantic-updates Release' does not have a Release file.
0.766 E: The repository 'http://archive.ubuntu.com/ubuntu mantic-backports Release' does not have a Release file.
```

maybe a rerun just fixes that? anyway that's weird

**Comment by tgross35** [comment]
> wow, is this stuff cursed or something?

I'm coming to the same conclusion. CI problems have been more prevalent in the past couple of weeks but usually it is s390x that fails.

Don't worry about it, I can retry individual jobs until they pass.

**Comment by tgross35** [comment]
Not just you, something docker-related must be wonky https://github.com/rust-lang/libc/pull/4116. I'll retry later today.

**Comment by tgross35** [comment]
I guess ubuntu cuts off repos after they go EOL, fixed in https://github.com/rust-lang/libc/pull/4120.

**Comment by tgross35** [comment]
I accidentally clicked "update with merge" rather than "update with rebase" and can't find any way to undo that. Could you rebase?

It might give me the option again once another PR updates `main`, will see...

**Comment by folkertdev** [comment]
sure, rebased and force pushed

**Comment by tgross35** [comment]
Thanks! Fingers crossed...

**Comment by folkertdev** [comment]
just out of curiosity: do we know why the spark CI action is that slow? (26min, the others complete <= 3 min). I suspect it's because the `test-runner-linux` script is used instead of a `qemu-sparc` invocation (that e.g. powerpc uses). It's not documented in the dockerfile why that approach is chosen. 

I'd investigate further but just can't get anything to even build, e.g.

```
> sh ci/run.sh sparc64-unknown-linux-gnu
+ mirrors_url=https://ci-mirrors.rust-lang.org/libc
+ target=sparc64-unknown-linux-gnu
+ [ -n  ]
+ [ sparc64-unknown-linux-gnu = s390x-unknown-linux-gnu ]
+ cargo test --no-default-features --manifest-path libc-test/Cargo.toml --target sparc64-unknown-linux-gnu
   Compiling libc v1.0.0-alpha.1 (/home/folkertdev/rust/libc)
   Compiling ctest2 v0.4.8
error[E0061]: this function takes 3 arguments but 2 arguments were supplied
    --> /home/folkertdev/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ctest2-0.4.8/src/lib.rs:2525:25
     |
2525 |                 Rc::new(macro_rules::compile(self.parse_sess, item)),
     |                         ^^^^^^^^^^^^^^^^^^^^                  ---- argument #2 of type `&RefCell<Features>` is missing
     |
note: function defined here
    --> /home/folkertdev/.cargo/registry/src/index.crates.io-6f17d22bba15001f/garando_syntax-0.1.0/src/ext/tt/macro_rules.rs:172:8
     |
172  | pub fn compile(sess: &ParseSess, features: &RefCell<Features>, def: &ast::Item) -> SyntaxExtension {
     |        ^^^^^^^
help: provide the argument
     |
2525 |                 Rc::new(macro_rules::compile(self.parse_sess, /* &RefCell<Features> */, item)),
     |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `ctest2` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

**Pull request merged**